<launch>

  <arg name="kinect_topic_name"    default="kinect_merge"/>
  <arg name="clusters_topic_name"  default="$(arg kinect_topic_name)/clusters" />
  <arg name="out_topic_name"       default="$(arg kinect_topic_name)/human_detection" />
  <arg name="downsampling"         default="true" /> <!-- false -->
  <arg name="voxel_size"           default="0.02" />   <!-- 0.05 -->
  <arg name="min_cluster_size"     default="300" />
  <arg name="clustering_tolerance" default="0.04" />  <!-- 2*voxel_size 0.06 -->
  <arg name="end_eff_frame"        default="ati_link" />
  <arg name="process_noise"        default="0.05" />
  <arg name="kinect_noise"         default="2.5" />
  <arg name="kinect_noise_z"       default="5" />
  <arg name="minimum_height"       default="0" />
  <arg name="max_tracking_jump"    default="0.35" />
  <arg name="tracking_type"        default="body/" />

  <node name="closest_b_pt_tracking" pkg="kinects_human_tracking" type="closest_pt_tracking" output="screen">
    <param name="kinect_topic_name"     value="$(arg kinect_topic_name)" />
    <param name="clusters_topic_name"   value="$(arg clusters_topic_name)" />
    <param name="out_topic_name"        value="$(arg out_topic_name)" />
    <param name="downsampling"          value="$(arg downsampling)" />
    <param name="voxel_size"            value="$(arg voxel_size)" />
    <param name="min_cluster_size"      value="$(arg min_cluster_size)" />
    <param name="clustering_tolerance"  value="$(arg clustering_tolerance)" />
    <param name="process_noise"         value="$(arg process_noise)" />
    <param name="kinect_noise"          value="$(arg kinect_noise)" />
    <param name="kinect_noise_z"        value="$(arg kinect_noise_z)" />
    <param name="minimum_height"        value="$(arg minimum_height)" />
    <param name="max_tracking_jump"     value="$(arg max_tracking_jump)" />
    <param name="end_eff_frame"         value="$(arg end_eff_frame)" />
    <param name="tracking_type"         value="$(arg tracking_type)" />

    <rosparam param="clipping_rules">[z, GT, 0.04, y, GT, -4.0, x, GT, -2.0, x, LT, 3.0]</rosparam>
  </node>

</launch>
